File: Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;
        char op[] = {'(', '{', '['};
        char cl[] = {')', '}', ']'};
        map<char, char> open;
        open.insert(pair<char,char>('(', ')'));
        open.insert(pair<char,char>('{','}'));
        open.insert(pair<char,char>('[',']'));
        if (s[0]==')' || s[0]=='}' || s[0]==']')
            return false;
        stack.push(s[0]);
        for (int i=1; i<s.length(); i++){
            if (s[i]==')' || s[i]=='}' || s[i]==']'){
                if (!stack.size())
                    return false;
                if (s[i]==open[stack.top()])
                    stack.pop();
                else
                    return false;
            }
            else
                stack.push(s[i]);
        }
        if (stack.size())
            return false;
        else
            return true;
    }
};

File: solutions/Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        op = {'(':')', '[':']', '{':'}'}
        cl = [')', '}', ']']
        if s[0] in cl:
            return False
        stack.append(s[0])
        for i in range(1, len(s)):
            if s[i] in cl:
                if len(stack)==0:
                    return False
                if s[i]==op[stack[-1]]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(s[i])
        if len(stack)==0:
            return True
        else:
            return False